### PR TITLE
Language switcher

### PR DESCRIPTION
**Pull request by peterMuriuki**
Add a language switcher component.
Tweak some styles to fix wierd overlap on the header.
Add the language switcher to the header component

### CODE REVIEW

**Comment by peterMuriuki** [comment]
switcher

**Comment by peterMuriuki** [comment]
looks like coveralls is down.

File: src/components/page/Header/Header.tsx [hunk 1]
```diff
@@ -58,20 +63,40 @@ export class HeaderComponent extends React.Component<HeaderProps, State> {
     }
 
     public render() {
+        const { i18n, t } = this.props;
+        const { authenticated, user } = this.props;
+
+        const languageOptions: LanguageOptions = {
+            en: 'English',
+            vi: 'Vietnamese',
+        };
```
Comment (moshthepitt) [inline src/components/page/Header/Header.tsx]: Should we leave this hard-coded here, or move it elsewhere? Thinking in terms of [Open/Closed](https://stackify.com/solid-design-open-closed-principle/)

File: src/components/page/Header/Header.tsx [hunk 2]
```diff
@@ -58,20 +63,40 @@ export class HeaderComponent extends React.Component<HeaderProps, State> {
...
+         * @param languageCode - contains the languageOption.key of the selected language option
+         */
+        const languageChangeHandler = (languageCode: LanguageCode) => {
+            i18n.changeLanguage(languageCode);
+        };
+
+        const languageSwitcherProps = {
+            onLanguageChange: languageChangeHandler,
+            allLanguageOptions: languageOptions,
+            supportedLanguages: ['en', 'vi'] as LanguageCode[],
```
Comment (moshthepitt) [inline src/components/page/Header/Header.tsx]: I'm not a fan of having so many instances of these magic strings in the code

File: src/components/LanguageSwitcher/index.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ * @param languageOptions - a map of all allowed language options
+ * @param supportedLanguages - an array of the keys for options that will be displayed
+ */
+function getSupportedLanguageOptions(languageOptions: LanguageOptions, supportedLanguages?: LanguageCode[]) {
+    const supported: LanguageOptions = {};
+    const supportedLangIsDefined = supportedLanguages && supportedLanguages.length > 0;
+    if (!supportedLangIsDefined) {
+        return supported;
+    }
+    Object.keys(languageOptions).forEach((languageCode) => {
```
Comment (machariamuguku) [inline src/components/LanguageSwitcher/index.tsx]: could probably use for...in to iterate languageOptions

File: src/components/LanguageSwitcher/index.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+/** globe icon with a dropdown where users can select language */
+import React from 'react';
+import { UncontrolledDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Fragment } from 'react';
+
+export type LanguageCode = 'en' | 'vi';
```
Comment (machariamuguku) [inline src/components/LanguageSwitcher/index.tsx]: these could probably be colocated where they are implemented

**Review by machariamuguku — DISMISSED** [review]

**Review by moshthepitt — APPROVED** [review]

**Review by kelvin-muchiri — APPROVED** [review]

**Pull request merged**